.github/workflows/pypi-upload.yaml

```diff
@@ -0,0 +1,32 @@
+on:
+  workflow_dispatch:
+  release:
+    types: [published]
+
+name: pypi-upload
+
+jobs:
+  pypi-upload:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v3
+
+    - name: Set up Python
+      uses: actions/setup-python@v3
+      with:
+        python-version: '3.x'
+
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install build
+
+    - name: Build package
+      run: python -m build
+
+    - name: Publish package
```

Comment:
**question (non-blocking):** Is there any reason not to use the [Trusted Publishing](https://github.com/pypa/gh-action-pypi-publish#trusted-publishing) model for this action rather than an API token?

Reply:
i'll check it out